[10.0][MIG][web_shortcut]

tarteo opened the pull request.

Commit:
Merge pull request #1 from OCA/10.0

merge

Commit:
Merge pull request #3 from OCA/10.0

merge

Commit:
Initial commit

[MIG] Migrated js

[MIG] Readme

[REM] console.log

[FIX] Removed using of 'select'

[FIX] Copyrights, disclaimer, authors

[FIX] web_shortcut: QWeb render doesn't use string keys
This fixes a compatibility issue in Odoo 10 Enterprise

[ADD] setup.py

[FIX] web_shortcut: bring compatibility with Odoo 10.0 Enterprise and fix duplicates bug
UserMenu is no longer used in Odoo 10.0 Enterprise, so we moved shortcut_menu from UserMenu to WebClient, using SystrayMenu which is common to Community and Enterprise versions.
Plus, the duplicate bug is fixed. Indeed, the shortcuts where sometimes displayed twice in the shortcut menu, in Enterprise and Community versions.

.eggs in gitaggregate

README

[FIX] Issue with module 'board'

[FIX] Toggling of button when page loaded

[FIX] Refactor to fix issues

[FIX] Toggling of button when page loaded

[FIX] Modal

[IMP] Translating shortcut menu items

Revert "[IMP] Translating shortcut menu items"

This reverts commit 12774b3093b1a44538562f20aee4dd5438fb3fdd.

Revert "[FIX] Modal"

This reverts commit 5089a5787dffa29d481ae96b316567a28665cc22.

Revert "[FIX] Toggling of button when page loaded"

This reverts commit fc107cf76674a68468f5cc65c0be123ed3f85482.

Revert "[FIX] Refactor to fix issues"

This reverts commit d54732716d975fba844bcd9aa966e416bcd8b970.

Revert "[FIX] Toggling of button when page loaded"

This reverts commit 4faa3bf6e09a1272bba41b54aa3752634612ab78.

[IMP] Changed default value of user_id to be up to date with the API

[FIX] web_shortcut: bug when adding from a top menu in Enterprise

[FIX] web_shortcut: find systray_menu in different place in Enterprise and Community

[MIG] Readme

[FIX] web_shortcut: QWeb render doesn't use string keys
This fixes a compatibility issue in Odoo 10 Enterprise

[ADD] setup.py

.eggs in gitaggregate

README

[FIX] Issue with module 'board'

[IMP] Translating shortcut menu items

Revert "[IMP] Translating shortcut menu items"

This reverts commit 12774b3093b1a44538562f20aee4dd5438fb3fdd.

[IMP] Changed default value of user_id to be up to date with the API

[FIX] web_shortcut: find systray_menu in different place in Enterprise and Community

Commit:
Initial commit

Commit:
Initial commit

[MIG] Migrated js

[MIG] Readme

[REM] console.log

[FIX] Removed using of 'select'

[FIX] Copyrights, disclaimer, authors

[FIX] web_shortcut: QWeb render doesn't use string keys
This fixes a compatibility issue in Odoo 10 Enterprise

[ADD] setup.py

[FIX] web_shortcut: bring compatibility with Odoo 10.0 Enterprise and fix duplicates bug
UserMenu is no longer used in Odoo 10.0 Enterprise, so we moved shortcut_menu from UserMenu to WebClient, using SystrayMenu which is common to Community and Enterprise versions.
Plus, the duplicate bug is fixed. Indeed, the shortcuts where sometimes displayed twice in the shortcut menu, in Enterprise and Community versions.

.eggs in gitaggregate

README

[FIX] Issue with module 'board'

[FIX] Toggling of button when page loaded

[FIX] Refactor to fix issues

[FIX] Toggling of button when page loaded

[FIX] Modal

[IMP] Translating shortcut menu items

Revert "[IMP] Translating shortcut menu items"

This reverts commit 12774b3093b1a44538562f20aee4dd5438fb3fdd.

Revert "[FIX] Modal"

This reverts commit 5089a5787dffa29d481ae96b316567a28665cc22.

Revert "[FIX] Toggling of button when page loaded"

This reverts commit fc107cf76674a68468f5cc65c0be123ed3f85482.

Revert "[FIX] Refactor to fix issues"

This reverts commit d54732716d975fba844bcd9aa966e416bcd8b970.

Revert "[FIX] Toggling of button when page loaded"

This reverts commit 4faa3bf6e09a1272bba41b54aa3752634612ab78.

[IMP] Changed default value of user_id to be up to date with the API

[FIX] web_shortcut: bug when adding from a top menu in Enterprise

Commit:
[MIG] Migrated js

Commit:
[MIG] Readme

Commit:
[REM] console.log

Commit:
[FIX] Removed using of 'select'

Commit:
[FIX] Copyrights, disclaimer, authors

ThomasBinsfeld committed:
[ADD] setup.py

adrienpeiffer commented on web_shortcut/models/web_shortcut.py:
```diff
@@ -0,0 +1,37 @@
+    menu_id = fields.Many2one('ir.ui.menu', ondelete='cascade')
+    user_id = fields.Many2one('res.users', 'User Ref.', required=True,
+                              ondelete='cascade', index=True,
+                              default=lambda obj, cr, uid, context: uid)
```
```lambda self: self.env.user.id```

JonathanNEMRY replied:
@tarteo Ok now, just the last tips of @adrienpeiffer  and GO for merge :smile:

tarteo replied:
Thanks @adrienpeiffer, I changed it :)

ThomasBinsfeld committed:
[FIX] web_shortcut: QWeb render doesn't use string keys
This fixes a compatibility issue in Odoo 10 Enterprise

Commit:
Merge pull request #4 from acsone/10.0-web_shortcut_fix_qweb_render_tbi

[FIX] web_shortcut: QWeb render doesn't use string keys

Commit:
Merge pull request #5 from acsone/10.0-web_shortcut_setup_tbi

[ADD] setup.py

ThomasBinsfeld committed:
[FIX] web_shortcut: bring compatibility with Odoo 10.0 Enterprise and fix duplicates bug
UserMenu is no longer used in Odoo 10.0 Enterprise, so we moved shortcut_menu from UserMenu to WebClient, using SystrayMenu which is common to Community and Enterprise versions.
Plus, the duplicate bug is fixed. Indeed, the shortcuts where sometimes displayed twice in the shortcut menu, in Enterprise and Community versions.

ThomasBinsfeld commented:
@tarteo Hi, thanks for the merges. I'm working on bringing the compatibility with Odoo 10.0 Enterprise. I'll submit a PR soon.

ThomasBinsfeld committed:
.eggs in gitaggregate

ThomasBinsfeld committed:
README

Commit:
Merge pull request #6 from acsone/10.0-web_shortcut_compatibility_enterprise_tbi

[MIG] [10.0] web_shortcut: bring compatibility with Odoo Enterprise

astirpe commented:
I'm experiencing an incompatibility with module "board".

Steps to reproduce:
- On a new database, install module `board`
- Open any view and add it to My Dashboard (click on Favourites)
- Install module `web_shortcut`
- Open My Dashboard, following error occurs:


```
Odoo Client Error

TypeError: this.action_manager.webclient is undefined

http://127.0.0.1:8069/web/content/271-b248ffe/web.assets_backend.js:2637
Traceback:
.shortcut_check@http://127.0.0.1:8069/web/content/271-b248ffe/web.assets_backend.js:2637:44
.switch_mode/<@http://127.0.0.1:8069/web/content/271-b248ffe/web.assets_backend.js:2635:2608
jQuery.Callbacks/fire@http://127.0.0.1:8069/web/content/251-9bf9793/web.assets_common.js:541:281
jQuery.Callbacks/self.fireWith@http://127.0.0.1:8069/web/content/251-9bf9793/web.assets_common.js:546:198
.Deferred/</deferred[tuple[0]]@http://127.0.0.1:8069/web/content/251-9bf9793/web.assets_common.js:548:31

...

jQuery.Callbacks/fire@http://127.0.0.1:8069/web/content/251-9bf9793/web.assets_common.js:541:281
jQuery.Callbacks/self.fireWith@http://127.0.0.1:8069/web/content/251-9bf9793/web.assets_common.js:546:198
done@http://127.0.0.1:8069/web/content/251-9bf9793/web.assets_common.js:937:86
.send/callback@http://127.0.0.1:8069/web/content/251-9bf9793/web.assets_common.js:957:15
```

pedrobaeza commented:
@Yajo, can you help @astirpe?

Commit:
[FIX] Issue with module 'board'

Commit:
[FIX] Toggling of button when page loaded

astirpe commented:
Yes, the incompatibility with module "board" is fixed to me. :+1:  Thanks @tarteo !

Commit:
[FIX] Refactor to fix issues

Commit:
[FIX] Toggling of button when page loaded

Commit:
[FIX] Modal

Commit:
[IMP] Translating shortcut menu items

yajo reviewed (approved):
Code review OK.

I never really understood why would anybody use this addon instead of just adding some favourites directly in the browser, 🤔 but I'm not gonna block it for that.

ThomasBinsfeld commented on web_shortcut/static/src/js/web_shortcut.js:
```diff
@@ -6,6 +6,7 @@ odoo.define('web.shortcut', function (require) {
         WebClient = require('web.WebClient'),
         ViewManager = require('web.ViewManager'),
         ActionManager = require('web.ActionManager'),
+        Menu = require('web.Menu'),
```
@tarteo That dependency makes the module unusable on Enterprise version.

tarteo replied:
@ThomasBinsfeld Sorry, I can't spend anymore time on this.

JonathanNEMRY replied:
hello @tarteo 
we spent a lot of time on this one too and with your last adding feature (https://github.com/OCA/web/pull/565/commits/4faa3bf6e09a1272bba41b54aa3752634612ab78) it's like reverting our contribution as this commit implies an inconsistency with enterprise version.
As you have no more time I suggest you to add this functionality into another module (event a local module to you) that inherit of this one. Many Thanks in advance

tarteo replied:
I reverted my changes to keep it compatible with Enterprise. My changes were to fix some bugs.

JonathanNEMRY replied:
Was it required to add web.Menu to fix those bugs? Can you point me the 'reintroduced' bug by this revert?

tarteo replied:
- When the page is refreshed on a favourite view the toggle is not 'checked'. And other cases that the toggle is not in the correct state.
- The toggle blinks every time the view changes

JonathanNEMRY reviewed (changes requested):
Do not use require('web.Menu') as it is not enterprise-compatible

Commit:
Revert "[IMP] Translating shortcut menu items"

This reverts commit 12774b3093b1a44538562f20aee4dd5438fb3fdd.

Commit:
Revert "[FIX] Modal"

This reverts commit 5089a5787dffa29d481ae96b316567a28665cc22.

Commit:
Revert "[FIX] Toggling of button when page loaded"

This reverts commit fc107cf76674a68468f5cc65c0be123ed3f85482.

Commit:
Revert "[FIX] Refactor to fix issues"

This reverts commit d54732716d975fba844bcd9aa966e416bcd8b970.

Commit:
Revert "[FIX] Toggling of button when page loaded"

This reverts commit 4faa3bf6e09a1272bba41b54aa3752634612ab78.

ThomasBinsfeld committed:
[FIX] web_shortcut: bug when adding from a top menu in Enterprise

ThomasBinsfeld committed:
[FIX] web_shortcut: find systray_menu in different place in Enterprise and Community

ThomasBinsfeld committed:
[FIX] web_shortcut: find systray_menu in different place in Enterprise and Community

Commit:
Merge pull request #8 from acsone/10.0-web_shortcut_fix_top_menus_tbi

[10.0] [REF] Web Shortcut: fix compatibility with Odoo Enterprise and Community

Commit:
[IMP] Changed default value of user_id to be up to date with the API

Commit:
Merge branch '100_mig_web_shortcut' of https://github.com/tarteo/web into 100_mig_web_shortcut

tarteo commented:
Can we merge?

pedrobaeza commented:
Can you please squash a bit your commits?

tarteo committed:
Merge branch '100_mig_web_shortcut' of https://github.com/tarteo/web into 100_mig_web_shortcut

tarteo commented:
@pedrobaeza done :)

pedrobaeza commented:
I still see 35 commits. Isn't this a lot for this module?

tarteo committed:
Merge branch '100_mig_web_shortcut' of https://github.com/tarteo/web into 100_mig_web_shortcut

tarteo commented:
@pedrobaeza I wasn't able to rebase, so i've created a new branch and pr. https://github.com/OCA/web/pull/626
